modules/GitterIRCBridge.js

```diff
@@ -0,0 +1,106 @@
+		bridge_config.cmd_prefix
+	);
+	
+	gitter.on("message", function(from, msg) {
+		msg = msg.replace(new RegExp(bridge_config.github.org + "/" + bridge_config.github.repo + "#", "gi"), "https://github.com/" + bridge_config.github.org + "/" + bridge_config.github.repo + "/issues/");
+		if (msg.match(/\[!\[(.*?)]/gi)) {
+			var img_name = msg.match(/\[!\[(.*?)]/gi)[0];
+			img_name = img_name.substring(3, img_name.length - 1);
+			var img_url = msg.match(/]\((.*?)\)]/gi)[0];
+			img_url = img_url.substring(2, img_url.length - 2);
+			irc.say(from, "shared '" + img_name + "': " + img_url);
+		} else {
+			irc.say(from, msg);
+		}
+	}).on("message_update", function(from, msg) {
+		msg = msg.replace(new RegExp(bridge_config.github.org + "/" + bridge_config.github.repo + "#", "gi"), "https://github.com/" + bridge_config.github.org + "/" + bridge_config.github.repo + "/issues/");
+		if (msg.match(/\[!\[(.*?)]/gi)) {
+			var img_name = msg.match(/\[!\[(.*?)]/gi)[0];
+			img_name = img_name.substring(3, img_name.length - 1);
+			var img_url = msg.match(/]\((.*?)\)]/gi)[0];
+			img_url = img_url.substring(2, img_url.length - 2);
+			irc.say(from, "[EDIT] shared '" + img_name + "': " + img_url);
+		} else {
+			irc.say(from, "[EDIT] " + msg);
+		}
+	});
+	irc.on("message", function(from, msg) {
+		gitter.say("`" + from + "` " + msg);
+	});
+	
+	irc.on("nick_changed", function(old_nick, new_nick) {
+		gitter.say("*" + old_nick + "* is now known as *" + new_nick + "*.")
+	}).on("join", function(nick) {
+		gitter.say("*" + nick + " has joined.*");
+	}).on("kick", function(nick, by, reason) {
+		gitter.say("*" + nick + " has been kicked by " + by + " for '" + reason + "'.*");
+	}).on("part", function(nick, reason) {
+		gitter.say("*" + nick + " has left.*");
+	}).on("quit", function(nick, reason, channels) {
+		gitter.say("*" + nick + " has quit.*");
+	}).on("kill", function(nick, reason, channels) {
+		gitter.say("*" + nick + " was killed by the IRC network for '" + reason + "'.*");
+	});
+	
+	irc.on("error", function(err) {
+		console.log(err);
+	});
+	gitter.on("error", function(err) {
+		console.log(err);
+	});
+	
+	for (var x in bridge_config.gitter.commands) {
+		gitter.on("command#" + x, function(command_data, from, args) {
+			if (command_data.not_elevated) {
+				command_data.callback(from, args, gitter, irc);
+			} else {
+				github.org(bridge_config.github.org).member(from, function(user, args, command, error, isOrgMember, headers) {
+					if(isOrgMember) {
+						command.callback(user, args, gitter, irc);
+					} else {
+						command.not_allowed(user, gitter, irc);
+					}
+				}.bind(this, from, args, command_data));
+			}
+		}.bind(this, bridge_config.gitter.commands[x]));
+	}
+	for (var y in bridge_config.irc.commands) {
+		irc.on("command#" + y, function(command_data, from, args) {
+			if (command_data.not_elevated) {
+				command_data.callback(from, args, irc, gitter);
+			} else {
+				if(irc.getUserRank(from) === "@") {
```

Comment:
Okay? I don't really have a preference either way but... its pretty trivial to figure out who's involved with the new xkit project haha